platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/admin/ExportMigrationConfigurationAdminEntry.java

```diff
@@ -55,7 +58,19 @@ public boolean store() {
     if (!stored) {
       LOGGER.debug(
           "Exporting configuration [{}] to [{}]...", configuration.getPid(), entry.getPath());
-      this.stored = entry.store((r, out) -> persister.write(out, configuration.getProperties()));
+      final Dictionary<String, Object> props = configuration.getProperties();
+
+      // just make sure the pid and factory pid are correct as sometimes they might get out of
+      // sync by an invalid update to the properties and we do rely on those at import time
+      final String factoryPid = configuration.getFactoryPid();
+
+      if (factoryPid != null) {
+        props.put(ConfigurationAdmin.SERVICE_FACTORYPID, factoryPid);
+      } else {
+        props.remove(ConfigurationAdmin.SERVICE_FACTORYPID);
+      }
+      props.put(Constants.SERVICE_PID, configuration.getPid());
+      this.stored = entry.store((r, out) -> persister.write(out, props));
```

Comment:
❗️We also need to handle the `ConfigurationAdmin.SERVICE_BUNDLELOCATION` / `getBundleLocation()` the same way here and in the import.

Reply:
nice! this was actually one of my question on the original PR. I didn't know wether or not the location should be restored. I will make these changes.